jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -263,7 +285,15 @@ private SpanContext createNewContext(String debugId) {
         }
       }
 
-      return new SpanContext(id, id, 0, flags);
+      // Check if active span should be established as CHILD_OF relationship
+      if (!ignoreActiveSpan && null != activeSpanSource.activeSpan()) {
+        SpanContext parentContext = (SpanContext) activeSpanSource.activeSpan().context();
+        asChildOf(parentContext);
+        traceId = parentContext.getTraceId();
+        parentId = parentContext.getSpanId();
+      }
+
+      return new SpanContext(traceId, id, parentId, flags);
```

Comment:
I think there's another issue - span ID can be unique or inherited from parent depending on  `zipkinSharedRpcSpan` flag. Looking at this with a fresh eye, perhaps this child logic does not belong to `createNewContext`. Instead, of we move it to the beginning of `startManual` would the rest just work automatically?

```java
    public io.opentracing.Span startManual() {
        // move here
        if (!ignoreActiveSpan && null != activeSpanSource.activeSpan()) {
               asChildOf((SpanContext) activeSpanSource.activeSpan().context());
        }

      SpanContext context;

      String debugId = debugId();
      if (references.isEmpty()) {
        context = createNewContext(null);
      } else if (debugId != null) {
        context = createNewContext(debugId);
      } else {
        context = createChildContext();
      }

```